### PR TITLE
fix: validate_string_identifier would allow some disallowed strings

### PR DESCRIPTION
**Pull request by Erk-**
Any string containing at least one valid character would be allowed by the regex, this fix removes regex and does a normal scan of the string instead.

This patch also contains some cleaning in the Cargo.toml file where it removes dependencies that was not in use.

### CODE REVIEW

**Comment by tommymalmqvist** [comment]
Hello again @Erk- sorry about the late response and thanks for the PR.

This is actually a remnant from when I had plans to implement a server and client in this repo and the file is not really used. I started writing a server in another repo.

Do you need it for something special?

**Comment by Erk-** [comment]
I don't need it for anything, I was just looking through the dependencies and was seeing if I could remove some of them and spotted the issue above. So it may be better to just remove it.

**Comment by tommymalmqvist** [comment]
Thanks I’ll remove it then!

**Comment by Erk-** [comment]
@tommymalmqvist I can see that it is used for the `#[validate(...)]` attributes, should these also be removed or would it make more sense to keep `validate_string_identifier` in that case?

**Comment by tommymalmqvist** [comment]
> @tommymalmqvist I can see that it is used for the `#[validate(...)]` attributes, should these also be removed or would it make more sense to keep `validate_string_identifier` in that case?

Yes, you are correct!

I could update the regex like so also:

```rust
let re = Regex::new(r"^[a-zA-Z0-9*+=:|@._-]*$").unwrap();
```

**Comment by Erk-** [comment]
Yeah that would also be a valid solution, though it should probably be `..]+$`.

Would you rather keep a regex based solution because then I can switch to that.

**Comment by tommymalmqvist** [comment]
Lets go with regex 👍

**Comment by Erk-** [comment]
@tommymalmqvist I have updated it to use regex, so you can have another look now.

File: src/v2_0_1/helpers/validator.rs [hunk 1]
```diff
@@ -1,19 +1,56 @@
...
 /// # identfierString
 /// This is a case-insensitive dataType and can only contain characters from the following
 /// character set: `a-z`, `A-Z`, `0-9`, `'*'`, `'-'`, `'_'`, `'='`, `':'`, `'+'`, `'|'`, `'@'`, `'.'`
 pub fn validate_identifier_string(s: &str) -> Result<(), ValidationError> {
     // regex for identifierString as defined by the specification
-    let re = Regex::new(r"[a-z]|[A-Z]|[0-9]|[+]|[*]|[-]|[=]|[:]|[0]|[|]|[@]|[.]").unwrap();
-
-    let res = re.is_match(s);
+    let res = REGEX
+        .get_or_init(|| Regex::new(r"^[a-zA-Z0-9*+=:|@._-]+$").unwrap())
```
Comment (tommymalmqvist) [inline src/v2_0_1/helpers/validator.rs:16]: The + symbol match the preceding character one or more times. In the specification the identifierString might be 0-n in length. I think we should use a * here instead.

For example in CostUpdatedRequest the specification of v2.0.1 states in the field type column: identifierString[0..36]

**Review by tommymalmqvist — CHANGES_REQUESTED** [review]

**Review by tommymalmqvist — APPROVED** [review]

**Pull request merged**